### PR TITLE
Fix SonarCloud analysis run on PR branch has wrong coverage on New Code

### LINKED ISSUE #2058: SonarCloud analysis run on PR branch has wrong `coverage on New Code`
Refer: https://github.com/apache/kvrocks/pull/2055/checks?check_run_id=20897911202

### PR DESCRIPTION
**Pull request by jihuayu**
Fix #2058

### CODE REVIEW

**Comment by PragmaTwice** [comment]
image

Actually I don't think there's any problem for sonarcloud to automatically detect the target and base branch.

**Comment by jihuayu** [comment]
> Actually I don't think there's any problem for sonarcloud to automatically detect the target and base branch.

I know now! I also remove the checkout step depth param, it works!
(Thank you for reminding me. I forgot that I also modified this parameter.)
image
https://sonarcloud.io/summary/new_code?id=jihuayu_kvrocks-test&pullRequest=1

**Comment by PragmaTwice** [comment]
IMHO the `fetch-depth: 0` SHOULD NOT be removed. Otherwise the blame information cannot be uploaded to sonarcloud.

```
WARN: Shallow clone detected, no blame information will be provided. You can convert to non-shallow with 'git fetch --unshallow'.
INFO: SCM Publisher 0/285 source files have been analyzed (done) | time=3ms
WARN: Missing blame information for the following files:
```

If you clone a repository with only one commit, Git cannot distinguish which lines belong to each author.

**Review by git-hulk — APPROVED** [review]

File: .github/workflows/sonar.yaml [hunk 1]
```diff
@@ -32,7 +32,6 @@ jobs:
         with:
           repository: ${{ github.event.workflow_run.head_repository.full_name }}
           ref: ${{ github.event.workflow_run.head_sha }}
-          fetch-depth: 0
```
Comment (PragmaTwice) [inline .github/workflows/sonar.yaml:35]: See above.

**Comment by jihuayu** [comment]
> IMHO the `fetch-depth: 0` SHOULD NOT be removed. Otherwise the blame information cannot be uploaded to sonarcloud.
> 
> ```
> WARN: Shallow clone detected, no blame information will be provided. You can convert to non-shallow with 'git fetch --unshallow'.
> INFO: SCM Publisher 0/285 source files have been analyzed (done) | time=3ms
> WARN: Missing blame information for the following files:
> ```
> 
> If you clone a repository with only one commit, Git cannot distinguish which lines belong to each author.

Ohh，I misunderstood the meaning of the "depth" parameter.
It seems there are still issues.

**Pull request closed without merging**